### PR TITLE
add host-local storage support to CnsVolumeCreateSpec and CnsPlacementResult

### PR DESCRIPTION
**Pull request by divyenpatel**
## Description

Add Hosts field to CnsVolumeCreateSpec for specifying target hosts when creating host-local datastore volumes, and add Host field to CnsPlacementResult to surface the selected host after placement. Add test coverage via the CNS_RUN_HOST_LOCAL_STORAGE_TESTS env var that validates host is non-nil in the placement result.

### CODE REVIEW

**Review by revathyshunmugam-ui — APPROVED** [review]

File: cns/client_test.go [hunk 1]
```diff
@@ -59,6 +59,11 @@ func TestClient(t *testing.T) {
 	// example: export CNS_RUN_MULTICLUSTER_PER_ZONE_TESTS='true'
 	run_cns_multicluster_per_zone_tests := os.Getenv("CNS_RUN_MULTICLUSTER_PER_ZONE_TESTS")
 
+	// set CNS_RUN_HOST_LOCAL_STORAGE_TESTS to true to run host-local datastore volume creation tests.
+	// The SPBM policy (CNS_SPBM_PROFILE_NAME) must have the hostLocalStorage capability.
+	// example: export CNS_RUN_HOST_LOCAL_STORAGE_TESTS='true'
+	run_host_local_storage_tests := os.Getenv("CNS_RUN_HOST_LOCAL_STORAGE_TESTS")
```
Comment (chethanv28) [inline cns/client_test.go:67]: In what version of vcenter/vsphere would this test be applicable ?
You can add a comment or add a version check in Line: 201

**Review by chethanv28 — DISMISSED** [review]
CreateVolume Spec changes look good to me. 

nit: Have a minor comment on the test code

File: cns/client_test.go (hunk 1 above)
Comment (divyenpatel) [reply cns/client_test.go:67]: target vCenter version is 9.1.2.
Addressed your review comment - https://github.com/vmware/govmomi/pull/4047/changes/66288bbb3a6b32fb276d723ca22762044d0f6ed0

**Review by dougm — APPROVED** [review]
